racket/src/cs/rumble/hash.ss

```diff
@@ -1065,6 +1116,17 @@
                             hash-procs-ref
                             ht k none))
 
+(define (impersonate-hash-ref-key ht k)
+  (impersonate-hash-ref/set 'hash-ref-key #f
+                            (lambda (ht k v) (hash-ref-key ht k none))
+                            (lambda (procs ht k none-v)
+                              (let-values ([(new-k _) (|#%app| (hash-procs-ref procs) ht k)])
+                                (values new-k
+                                        (lambda (ht k none-v)
```

Comment:
This lambda requires a closure  (as `procs` is bound by the enclosing lambda). Not sure if that's okay.

Also, if `ht` is a hash chaperone and the resulting key isn't `chaperone-of?` the original, the error message will describe the key as a "value":
```racket
                   [else
                    (let ([new-r (new-v-or-wrap next-ht new-k r)])
                      (when chaperone?
                        (unless (chaperone-of? new-r r)
                          (raise-chaperone-error who "value" new-r r)))
                      new-r)]
```

To correct that, maybe we should add a `what-r` parameter to `impersonate-hash-ref/set`?

Reply:
Adding a `what-r` to `impersonate-hash-ref/set` sounds like a good idea.

Reply:
Fixed in 42425bb70de068932e0aaced5d133fe7583f8b76